examples/release_highlights/plot_release_highlights_0_24_0.py

```diff
@@ -0,0 +1,225 @@
+# explore the space of the parameters and identify their best combination.
+# It is still experimental, in order to import it `enable_halving_search_cv`
+# should be imported from `experimental`.
+# :class:`~sklearn.model_selection.HalvingGridSearchCV` and
+# :class:`~sklearn.model_selection.HalvingRandomSearchCV` can be
+# used as drop-in replacement for
+# :class:`~sklearn.model_selection.GridSearchCV` and
+# :class:`~sklearn.model_selection.RandomizedSearchCV`.
+# Successive halving is an iterative selection process. The first iteration is
+# run with a small amount of resources, by resource meaning typically the
+# number of training samples, but also arbitrary numeric parameters such
+# as `n_estimators` in a random forest. Only some of the parameter candidates
+# are selected for the next iteration, with an increasing size of the
+# allocated resources.
+# As illustrated in the figure below,
+# only a subset of candidates will last until the end of the iteration process.
+# Read more in the :ref:`User Guide <successive_halving_user_guide>`.
+# 
+# .. figure:: ../model_selection/images/sphx_glr_plot_successive_halving_iterations_001.png
+#   :target: ../model_selection/plot_successive_halving_iterations.html
+#   :align: center
+
+import numpy as np
+from scipy.stats import randint
+from sklearn.experimental import enable_halving_search_cv  # noqa
+from sklearn.model_selection import HalvingRandomSearchCV
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.datasets import make_classification
+
+rng = np.random.RandomState(0)
+
+X, y = make_classification(n_samples=700, random_state=rng)
+
+clf = RandomForestClassifier(n_estimators=20, random_state=rng)
+
+param_dist = {"max_depth": [3, None],
+              "max_features": randint(1, 11),
+              "min_samples_split": randint(2, 11),
+              "bootstrap": [True, False],
+              "criterion": ["gini", "entropy"]}
+
+rsh = HalvingRandomSearchCV(estimator=clf, param_distributions=param_dist,
+                            factor=2, random_state=rng)
+rsh.fit(X, y)
+rsh.best_params_
+
+##############################################################################
+# Native support for categorical features in HistGradientBoosting
+# ---------------------------------------------------------------
+# :class:`sklearn.ensemble.HistGradientBoostingClassifier` and
+# :class:`sklearn.ensemble.HistGradientBoostingRegressor` now have native
+# support for categorical features: they can consider splits on non-ordered,
+# categorical data. Read more in the :ref:`User Guide
+# <categorical_support_gbdt>`.
+#
+# .. figure:: ../ensemble/images/sphx_glr_plot_gradient_boosting_categorical_001.png
+#   :target: ../ensemble/plot_gradient_boosting_categorical.html
+#   :align: center
+
+##############################################################################
+# Improved performances in HistGradientBoosting methods
```

Comment:
Not a highlight IMHO, as it's probably too specific. Also the main improvement wasn't fitting time, but memory usage.

Reply:
I also tend to exclude this from highlighting.

Reply:
I think it's really worth highlighting that the memory efficiency has been greatly improved, typically by a factor of 10x or more on models with many features and deep trees. This is a game changer for many potential users that would have had no choice but to stick with lightgbm or xgboost otherwise.

It's also a good opportunity to highlight the continuous benchmarking report that is build daily with ASV.

I agree the speed performance improvement for this model is secondary.